src/System.AppContext/src/System/AppContext.cs

```diff
@@ -18,8 +18,7 @@ private enum SwitchValueState
             UnknownValue = 0x8 // Not used on .NET Native
         }
 
-        private static Dictionary<string, SwitchValueState> s_switchMap = new Dictionary<string, SwitchValueState>();
-        private static readonly object s_syncLock = new object();
+        private static readonly Dictionary<string, SwitchValueState> s_switchMap = new Dictionary<string, SwitchValueState>();
```

Comment:
> This could simplified by using bool values

Or a `HashSet<string>`. Or even a `string[]` if the number of entries is low enough (I suspect it may be).